docs/archapi.md

```diff
@@ -88,14 +88,18 @@ Get Z dimension for the specified tile for bels. All bels with at specified X an
 
 Get Z dimension for the specified tile for pips. All pips with at specified X and Y coordinates must have a Z coordinate in the range `0 .. getTileDimZ(X,Y)-1` (inclusive).
 
+### char getNameDelimiter() const
```

Comment:
I have a suggestion here, but it does add complexity.  Have the `IdStringList` have a `DelimiterType` field.  A new archapi method could look something like:
```
const_range<IdString> getDelimiters(DelimiterType delimiters) const
```

This could enable different IdStringList's to have different presentations.  This would allow an arch to break down as needed, examples:
```
Pip: $tile/$wire0->>$wire1 ==> getDelimiters() => ["/", "->>"]
SitePip: $site.$site_type/$bel/$bel_pin ==> getDelimiters() => [".", "/", "/"]
```

This can get dangerous if misused, and I'm a little worried about complexity and / or IdStringList size bloat.  However I think it answers a TODO you had over here: https://github.com/YosysHQ/nextpnr/pull/561/files#diff-effe03ae96858aeff9455ed9406ec25b61ff5a52122abb6b3cb49b16f51b0dd0R314

Reply:
I need to think about this; the bigger problem I think is the complexity that it would add to the parsing path.

Reply:
Don't disagree about complexity.  But I do think the runtime cost can be managed.

Reply:
So at the moment we can parse a string into an `IdStringList` with the only needed context being the current architecture. Doing it with variable delimiters would potentially require more context in the parsing, particularly if you want to support things like different delimiters inside different pip types.

Reply:
A DFA (discrete finite automata) could be generated pretty easily to do the parsing without additional context.

Reply:
I have thought about this a bit more, and while it's a nice idea given the feedback I have had about the Arch API being too complex already I think it is simply too much extra complexity for the benefit right now.